deps/src/singular.cpp

```diff
@@ -54,6 +54,84 @@ static void WerrorS_and_reset(const char * s)
   singular_error_log.emplace_back(s);
 }
 
+#define SETTER(A, B)                                                                     \
+  else if (opt == #B)                                                                    \
+  {                                                                                      \
+    old_value = (A & Sy_bit(B)) != 0;                                                    \
+    A = value ? (A | Sy_bit(B)) : (A & ~Sy_bit(B));                                      \
+  }
+
+// all of the global setters return the previous value
+auto set_option_helper(std::string opt, bool value, ring r) {
+    bool old_value = false;
+    ring oldring=currRing;
+    if (r!=NULL) rChangeCurrRing(r);
+    if (false)
+      ;
```

Comment:
```suggestion
      ;
```
what is this for?

Reply:
needed to apply the SETTER macro: "else if ...."